Sources/Swift/Helper/SentryNSTimerFactory.swift

```diff
@@ -0,0 +1,25 @@
+import Foundation
+
+@objc
+@_spi(Private) public class SentryNSTimerFactory: NSObject {
+    
+    @objc @discardableResult
+    @_spi(Private) public func scheduledTimer(withTimeInterval interval: TimeInterval, repeats: Bool, block: @escaping (Timer) -> Void) -> Timer {
+        if !Thread.isMainThread {
+            let warningText = "Timers must be scheduled from the main thread, or they may never fire. See the attribute on the declaration in NSTimer.h. See https://stackoverflow.com/questions/8304702/how-do-i-create-a-nstimer-on-a-background-thread for more info."
+            SentrySDKLog.warning(warningText)
+            assert(false, warningText)
```

Comment:
`l`: nit-pick, can be simplified with [assertionFailure(_:)](https://developer.apple.com/documentation/swift/assertionfailure(_:file:line:))

```suggestion
            assertionFailure(warningText)
```